### PR TITLE
Refactor tractor cjk

### PR DESCRIPTION
**Pull request by ckaminer**
What's this PR do?
- This PR covers the first crack at refactoring our board module.

Where should the reviewer start?
- The file comparison will probably be very useful for reviewing this PR.  The logic hasn't changed much.

How should this be manually tested?
- Pull down the project, fire up `npm start` and try playing!

Any background context you want to provide?
- We have nodes and lists that are specific to directions.  Originally there were separate functions for each direction.  The purpose of this refactoring is to pull the logic into one function that takes direction as an argument.

Screenshots (if appropriate)

What gif best describes this PR or how it makes you feel?
[Looking back at what we had](http://www.reactiongifs.com/r/idjut.gif)

### CODE REVIEW

**Comment by ckaminer** [comment]
@neight-allen @lucyfox4131

**Comment by icorson3** [comment]
looks great. nice refactoring.

File: lib/board.js [hunk 1]
```diff
@@ -53,99 +53,56 @@ function setupNodeGrid(nodes){
...
-  if (leftIds.indexOf(node.id) > -1 ) {
+  } else if (direction === "right") {
     node.rightNode = "edge";
-  } else {
-    var currentNodeIndex = nodes.indexOf(node);
-    node.rightNode = nodes[currentNodeIndex + 1];
   }
 }
 
-function createUpLists(nodes) {
```
Comment (lucyfox4131) [inline lib/board.js:103]: Looks like a great refactor here to take away repetitive code.

Comment (ckaminer) [reply lib/board.js:103]: Thanks! Similar opportunities appear throughout our code so I was glad to get a start on it.  I tried looking into some metaprogramming-ish but had no luck.  Thought this was a decent start.

**Pull request merged**